python/cloudtik/providers/_private/aliyun/node_provider.py

```diff
@@ -266,18 +265,24 @@ def create_node(
                         tags=tags_for_creation,
                         count=count
                     )
-                    instances = self.ecs.describe_instances(instance_ids=instance_id_sets)
-                    with cli_logger.group(
-                            "Launched {} nodes", count, _tags=cli_logger_tags):
-                        for instance in instances:
-                            created_nodes_dict[instance.instance_id] = instance
-                            cli_logger.print(
-                                "Launched instance {}",
-                                instance.instance_id,
-                                _tags=dict(
-                                    type=instance.instance_type,
-                                    status=instance.status))
-                        break
+                    # Make sure the status of all instances running before return all created nodes info.
+                    while True:
+                        instances = self.ecs.describe_instances(instance_ids=instance_id_sets, status=RUNNING)
+                        if instances and len(instances) == len(instance_id_sets):
+                            with cli_logger.group(
+                                    "Launched {} nodes", count, _tags=cli_logger_tags):
+                                for instance in instances:
+                                    created_nodes_dict[instance.instance_id] = instance
+                                    cli_logger.print(
+                                        "Launched instance {}",
+                                        instance.instance_id,
+                                        _tags=dict(
+                                            type=instance.instance_type,
+                                            status=instance.status))
+                            break
+                        cli_logger.error("Waiting for all the status of created instances running.")
+                        time.sleep(STARTING_NODE_DELAY)
+                    break
```

Comment:
Thanks for the fix. I made a few improvements based on your code. Please check below:
1. We cannot loop forever for robust. So we have a max attempts. If if the status checked failed with max attempts, it doesn't matter, we can go on. 
2. We don't need to wait to RUNNING which usually takes quite some time to reach this status. We just need them be starting status which may be much faster to reach.


MAX_CREATE_STATUS_ATTEMPTS = 30
STARTING_NODE_DELAY = 1

                    status_attempt = 0
                    while status_attempt < MAX_CREATE_STATUS_ATTEMPTS:
                        instances = self.ecs.describe_instances(instance_ids=instance_id_sets)
                        if instances:
                            # Counting on both STARTING and RUNNING nodes
                            started_nodes = 0
                            for instance in instances:
                                if instance.status == STARTING or instance.status == RUNNING:
                                    started_nodes += 1
                            if started_nodes == len(instance_id_sets):
                                # All in starting or running status
                                with cli_logger.group(
                                        "Launched {} nodes", count, _tags=cli_logger_tags):
                                    for instance in instances:
                                        created_nodes_dict[instance.instance_id] = instance
                                        cli_logger.print(
                                            "Launched instance {}",
                                            instance.instance_id,
                                            _tags=dict(
                                                type=instance.instance_type,
                                                status=instance.status))
                                break
                        if status_attempt == MAX_CREATE_STATUS_ATTEMPTS - 1:
                            # Even if the status check failed, we simply show a warning
                            # Because this doesn't really prevent anything
                            cli_logger.warning(
                                "Status checking for node creation failed after {} attempts.", status_attempt)
                            break
                        time.sleep(STARTING_NODE_DELAY)
                    break